demo.py

```diff
@@ -183,7 +189,36 @@ def register(conn, users):
             if args.reg_notify:
                 im = imessage.iMessageUser(conn, email_user)
                 im.current_handle = email_addr
-                await im.send(imessage.iMessage.create(im, "Number registration is valid until " + expiration, [email_addr]))
+                await im.send(imessage.iMessage.create(im, "Number registration is valid until " + expirationstr, [email_addr]))
+
+            if args.daemon:
+
+                wait_time_minutes = 5  # this is in minutes. 5 recommended
+
+                async def trigger_event(event_time, seconds):
+                    current_time = datetime.datetime.utcnow()
+                    time_difference = event_time - current_time
+                    seconds_until_event = int(time_difference.total_seconds())
+
+                    if seconds_until_event <= 0:
+                        print("You are not registered. Reregistering...")
+                    elif seconds_until_event <= seconds + 10:
+                        print("You are set to be deregistered in less than 5 minutes. Reregistering now...")
+                    else:
+                        print(f"Waiting for {int(seconds_until_event / 60)} minutes...")
+                        await trio.sleep(seconds_until_event)
+                        # time.sleep(10)
+                        print("Reregistered.")
+
+                while True:
+                    wait_time_seconds = wait_time_minutes * 60
```

Comment:
I would just replace this with something much simpler:
Don't really need to bring trigger_event out into a separate function.
```suggestion
                    reregister_time = expiration - datetime.timedelta(minutes=wait_time_minutes) # wait_time_minutes before expiration
                    reregister_delta = (reregister_time - datetime.datetime.utcnow()).total_seconds()
                    logging.info(f"Reregistering in {int(reregister_delta / 60)} minutes...")
                    await trio.sleep(reregister_delta)
                    logging.info("Reregistering...")
                    register(conn, users)
                    logging.info("Reregistered!")
```

Reply:
Also, I would be careful with UTC times. I just noticed that above we are using `cert.not_valid_after`, pretty sure that is deprecated in favor of `cert.not_valid_after_utc`

Reply:
Good call, just fixed this